app/components/DataCard.tsx

```diff
@@ -47,10 +52,18 @@ export default function DataCard<T extends Record<string, unknown>>({
     });
 
   const onSave = async (value: unknown) => {
+    if (!editColumn) {
+      // not possible
+      return;
+    }
+    // Save the value to the foreign key column, if exists
+    const saveColumnName = editColumn.relationship?.foreignKey
+      ? editColumn.relationship?.foreignKey
+      : editColumn.path;
```

Comment:
this hints at the awkward dance we need to do between _rendering_ a value from an object relationship. E.g. `crash. collsn.label` vs. editing that value, which requires us to update the foreign key at `crash.fhe_collsn_id`.

This PR is trying to iron this out by moving the necessary metadata into the column `relationship` config, whereas before we had some special lookup table logic hardcoded into various functions. 

One important thing that these changes do is open up future iterations where we are editing relationships that are not just plain old lookup tables. E.g., if we wanted a user to select an EMS record from a list so that it can be linked to a crash.

Reply:
ah, right—one other wrinkle that is not new to this PR is that in order to be able to edit lookup values our graphql queries need to include both the foreign key column and the related record itself.

For example, in the `Crash` query, you can see [here](https://github.com/cityofaustin/vision-zero/blob/7e95be2e3264d6bae0f3aeacdab0f9a9bf701bd2/app/queries/crash.ts#L20-L24) that we have

```graphql
     # this is the foreign key to the lookups.collsn table
      fhe_collsn_id
      # this is the related collision type lookup value
      collsn {
        id
        label
      }
```

If you were to omit the foreign key from the query, the `select` input in the UI would never correctly show the currently selected lookup option. So it would be fairly obvious that something was wrong. This is not documented anywhere ☹️